src/run-migration.js

```diff
@@ -21,27 +29,21 @@ module.exports = client => migration => {
     ? () => client.query("ROLLBACK")
     : noop
 
-  return bluebird
-    .resolve()
-    .then(begin)
-    .then(() => client.query(migration.sql))
-    .then(() => {
-      return client.query(
-        SQL`
-        INSERT INTO migrations (id, name, hash)
-          VALUES (${migration.id}, ${migration.name}, ${migration.hash})
-      `
-      )
-    })
-    .then(end)
-    .catch(err => {
-      return bluebird.resolve().tap(cleanup).then(() => {
-        throw new Error(
-          dedent`
+  try {
+    await begin()
+    await client.query(migration.sql)
+    await insertMigration(migrationTableName, client, migration, log)
+    await end()
+
+    return migration
+  } catch (err) {
+    await cleanup()
+
+    throw new Error(
```

Comment:
I reckon this should be in a `finally` block - we want to throw this error even if `cleanup` throws.